ts_src/payments/p2air.ts

```diff
@@ -0,0 +1,178 @@
+import { bitcoin as BITCOIN_NETWORK } from '../networks';
+import * as bscript from '../script';
+import * as vaultair from '../templates/vaultair';
+import { Payment, PaymentOpts, Stack, VaultTxType } from './index';
+import * as lazy from './lazy';
+
+const OPS = bscript.OPS;
+const typef = require('typeforce');
+const ecc = require('tiny-secp256k1');
+
+const OP_INT_BASE = OPS.OP_RESERVED; // OP_1 - 1
+
+function stacksEqual(a: Buffer[], b: Buffer[]): boolean {
+  if (a.length !== b.length) return false;
+
+  return a.every((x, i) => {
+    return x.equals(b[i]);
+  });
+}
+
+// input: OP_0 signature_1 [signature2, signature3] [OP_0/0x01] OP_0/0x01
+// output: OP_IF 1 OP_ELSE OP_IF 2 OP_ELSE 3 OP_ENDIF OP_ENDIF pubKey1 pubKey2 pubKey3 3 OP_CHECKMULTISIG
+export function p2air(a: Payment, opts?: PaymentOpts): Payment {
+  if (!a.input && !a.output && !a.pubkeys && !a.signatures)
+    throw new TypeError('Not enough data');
+  if (a.vaultTxType === undefined) a.vaultTxType = VaultTxType.Alert;
+
+  opts = Object.assign({ validate: true }, opts || {});
+
+  function isAcceptableSignature(x: Buffer | number): boolean {
```

Comment:
function isAcceptableSignature(x: Buffer | number): boolean {
    if (typeof x === 'number') {
      return (opts!.allowIncomplete && (x) === OPS.OP_0) !== undefined
    }
    return bscript.isCanonicalScriptSignature(x);
  }

would be safer, using "as" is kind of tricking typescript because you take responsibility that you know what you are doing and the types would as you declared